mithril-aggregator/src/artifact_builder/cardano_database_artifacts/immutable.rs

```diff
@@ -0,0 +1,788 @@
+                    PathBuf::from("/tmp/cardano-database/immutable/00002.tar.gz"),
+                ],
+            )
+        }
+
+        #[test]
+        fn return_error_when_one_of_the_three_immutable_files_is_missing() {
+            let test_dir =
+                "error_when_one_of_the_three_immutable_files_is_missing/cardano_database";
+            let cardano_db = DummyCardanoDbBuilder::new(test_dir)
+                .with_immutables(&[1, 2])
+                .build();
+
+            let file_to_remove = cardano_db.get_immutable_dir().join("00002.chunk");
+            std::fs::remove_file(file_to_remove).unwrap();
+
+            let db_directory = cardano_db.get_dir().to_path_buf();
+            let mut snapshotter = CompressedArchiveSnapshotter::new(
+                db_directory.clone(),
+                db_directory.parent().unwrap().join("snapshot_dest"),
+                SnapshotterCompressionAlgorithm::Gzip,
+                TestLogger::stdout(),
+            )
+            .unwrap();
+            snapshotter.set_sub_temp_dir(Uuid::new_v4().to_string());
+
+            let builder = ImmutableArtifactBuilder::new(
+                vec![Arc::new(MockImmutableFilesUploader::new())],
+                Arc::new(snapshotter),
+                CompressionAlgorithm::Gzip,
+                TestLogger::stdout(),
+            )
+            .unwrap();
+
+            builder
+                .immutable_archives_paths_creating_the_missing_ones(2)
+                .expect_err(
+                    "Should return an error when one of the three immutable files is missing",
+                );
+        }
+
+        #[test]
+        fn return_error_when_an_immutable_file_trio_is_missing() {
+            let test_dir = "error_when_an_immutable_file_trio_is_missing/cardano_database";
+            let cardano_db = DummyCardanoDbBuilder::new(test_dir)
+                .with_immutables(&[1, 3])
+                .build();
+
+            let db_directory = cardano_db.get_dir().to_path_buf();
+            let mut snapshotter = CompressedArchiveSnapshotter::new(
+                db_directory.clone(),
+                db_directory.parent().unwrap().join("snapshot_dest"),
+                SnapshotterCompressionAlgorithm::Gzip,
+                TestLogger::stdout(),
+            )
+            .unwrap();
+            snapshotter.set_sub_temp_dir(Uuid::new_v4().to_string());
+
+            let builder = ImmutableArtifactBuilder::new(
+                vec![Arc::new(MockImmutableFilesUploader::new())],
+                Arc::new(snapshotter),
+                CompressionAlgorithm::Gzip,
+                TestLogger::stdout(),
+            )
+            .unwrap();
+
+            builder
+                .immutable_archives_paths_creating_the_missing_ones(3)
+                .expect_err("Should return an error when an immutable file trio is missing");
+        }
+
+        #[test]
+        fn return_error_when_up_to_immutable_file_number_is_missing() {
```

Comment:
Maybe this name is more clear?
```suggestion
        fn return_error_when_immutable_file_number_is_not_produced_yet() {
```

Reply:
Done